pkg/channel/lookup.go

```diff
@@ -28,14 +29,29 @@ type ModuleTemplate struct {
 
 type ModuleTemplatesByModuleName map[string]*ModuleTemplate
 
-func GetTemplates(
-	ctx context.Context, client client.Reader, kyma *operatorv1beta1.Kyma,
+func GetTemplates(ctx context.Context, kymaClient client.Reader, kyma *operatorv1beta1.Kyma,
 ) (ModuleTemplatesByModuleName, error) {
 	logger := ctrlLog.FromContext(ctx)
 	templates := make(ModuleTemplatesByModuleName)
 
+	var runtimeClient client.Reader
+	if kyma.Spec.Sync.Enabled {
+		runtimeClient = remote.SyncContextFromContext(ctx).RuntimeClient
```

Comment:
If `kyma.Spec.Sync.Enabled` is not enabled, `runtimeClient` will be nil.

Here we should make a decision, if sync is not enabled, then this `RemoteModuleTemplateRef` should not be used, IMO.

Reply:
good idea. I thought of this but it felt like it's an very unlikely edge case.
But let's just be sure and check for the sync when using the RemoteModuleTemplateRef.